src/main/java/software/amazon/smithy/lsp/ext/model/SmithyBuildExtensions.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.lsp.ext.model;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import software.amazon.smithy.lsp.ext.LspLog;
+import software.amazon.smithy.utils.ListUtils;
+import software.amazon.smithy.utils.SmithyBuilder;
+import software.amazon.smithy.utils.ToSmithyBuilder;
+
+public final class SmithyBuildExtensions implements ToSmithyBuilder<SmithyBuildExtensions> {
```

Comment:
Why is this an LSP feature and not part of smithy-build?

Reply:
What API guarantees do we make in this repo? Can we yank this all out when it's moved to smithy-build?

Reply:
On the IDE/Editor client side, there will just be a watcher that detects changes to smithy-build.json and notifies the server of updates to the file. No one should need to use SmithyBuildExtensions directly.